container/array/array_test.go

```diff
@@ -0,0 +1,178 @@
+package array
+
+import (
+	"testing"
+	"unsafe"
+)
+
+const testLen = 100000
+
+func TestCapacity(t *testing.T) {
+	var a Array[int]
+	for i := 0; i < testLen; i++ {
+		a.Push(i)
+	}
+
+	if a.Len() != testLen {
+		t.Errorf("incorrect length: got=%d, expect=%d", a.Len(), testLen)
+	}
+
+	if expectCap := BlockCap[int](testLen); a.Cap() != expectCap {
+		t.Errorf("incorrect capacity: got=%d, expect=%d", a.Cap(), expectCap)
+	}
+
+	for i := 0; i < testLen; i++ {
+		if a.Get(i) != i {
+			t.Errorf("incorrect value at %d: got=%d, expect=%d", i, a.Get(i), i)
+		}
+	}
+
+	for i := 0; i < testLen; i++ {
+		a.Pop()
```

Comment:
Assert the value is correct here?

Reply:
I do in other tests, but yeah, that could be verified here too.

Reply:
Oops, I didn't notice that! Up to you if you want to adjust the behavior to return the value popped. Otherwise can keep the tests as is.

Reply:
Ok, I've returned the value during Pop and added a test for the value.